src/Parallel/Algorithm.hpp

```diff
@@ -317,7 +386,8 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
 
   // @{
   /// Start evaluating the algorithm until the is_ready function of an Action
-  /// returns false, or an Action returns with `terminate` set to `true`
+  /// returns false, or an Action returns with `terminate` set to `true` or
+  /// returns a different phase to be executed next.
```

Comment:
Big picture discussion: if an action says "next time we get to a global sync, we'll need to checkpoint", why should that interrupt the flow of the algorithm?

Do I misunderstand what you wrote here? I thought we wanted (in this first-pass implementation) not to have any requests that interrupt flow.

Reply:
You're correct -- I just failed to correct this comment after one of the code iterations, so I'll adjust it to reflect the current state.